client/lib/site/index.js

```diff
@@ -61,14 +61,16 @@ Site.prototype.attributes = function( attributes ) {
 };
 
 Site.prototype.set = function( attributes ) {
-	var changed = false;
+	let changed = false;
+	const computedAttributes = Object.assign( {}, attributes, getAttributes( attributes ) );
 
-	for ( var prop in attributes ) {
-		if ( attributes.hasOwnProperty( prop ) && ! isEqual( attributes[ prop ], this[ prop ] ) ) {
-			if ( undefined === attributes[ prop ] ) {
+	// Loop only through the original attributes, use computed values for comparison
```

Comment:
Actually, now that I think about this more, could we iterate over `computedAttributes` instead? `this` should contain computed attributes too, so I don't see why we would iterate over `attributes`.